include/circt/Dialect/Seq/SeqOps.td

```diff
@@ -324,6 +324,32 @@ def ClockGateOp : SeqOp<"clock_gate", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Clock Multiplexer
+//===----------------------------------------------------------------------===//
+
+def ClockMuxOp : SeqOp<"clock_mux", [Pure]> {
+  let summary = "Safely selects a clock based on a condition";
+  let description = [{
+    The `seq.clock_mux` op selects a clock from two options based on a
+    condition without glitches. If `cond` is true, the first clock operand
+    is selected to drive downstream logic. Otherwise, the second clock is used.
```

Comment:
I would remove the `without glitches` part.

In ASIC technologies, the dedicated clock mux cells are almost always just regular mux cells, but with the 0-to-1 rise and 1-to-0 fall time of their output balanced, while regular mux cell outputs fall faster than they rise. (PMOS transistors are generally weaker/slower than NMOS; ASIC stdcells have larger PMOS than NMOS to compensate a bit; regular muxes don't compensate fully, so rise time is usually slower than fall time; clock muxes trade cell area for providing balanced rise and fall times.)

To provide glitchless muxing, you'd generally combine a clock mux with a clock gate cell and have a state machine that first gates the clock, switches the mux once it's safe, and then ungates it again. Same goes for clock XOR/AND/OR gates. Mostly about balanced rise/fall times.